fc/parsing/actions.py

```diff
@@ -1452,6 +1474,128 @@ def get(ref):
+                ns_uri = self.ns_map[prefix]
+                try:
+                    var = self.model.get_variable_by_ontology_term((ns_uri, local_name))
+                except KeyError:
+                    raise ProtocolError(
+                        'Variables appearing in unit conversion rules must be either local variables or annotated'
+                        ' variables that already exist in the model.')
+
+                # Check that this variable won't be unit converted
+                pvar = model_var_to_pvar.get(var, None)
+                if pvar is not None and pvar.units is not None:
+                    units = self.units.get_unit(pvar.units)
+                    if units != var.units:
+                        raise ProtocolError(
+                            f'Variables appearing in unit conversion rules cannot themselves be unit-converted'
+                            f' ({pvar.long_name} is not currently in {pvar.units}).')
+
+                # Create and return pint quantity
+                return self.units.Quantity(var, var.units)
+
+            # Local variable
+            try:
+                return self.local_vars[name]
+            except KeyError:
+
+                # Try creating: must have explicit units
+                units = None
+                pvar = local_var_to_pvar.get(name, None)
+                if pvar is None or pvar.units is None:
+                    raise ProtocolError(
+                        'Local variables appearing in unit conversion rules must explicitly define their units.')
+
+                # Create, store, and return
+                units = self.units.get_unit(pvar.units)
+                name = self.model.get_unique_name('protocol__' + pvar.short_name)
+                pvar.model_variable = self.model.add_variable(name, units)
+                self.local_vars[pvar.name] = pvar.model_variable
+
+                # Create and return pint quantity
+                return self.units.Quantity(pvar.model_variable, units)
+
+        for rule in self.unit_conversion_rules:
+
+            # Get from and to units
+            u1 = self.units.get_unit(rule.actual)
+            u2 = self.units.get_unit(rule.desired)
+
+            # Get argument name and lambda expression
+            f = rule.lambda_function
+            if len(f.formal_parameters) != 1:
+                raise ProtocolError('A unit conversion rule must be a lambda function with a single argument.')
+            expr = f.body
+            free = f.formal_parameters[0]
+
+            # Convert expression to sympy
+            try:
+                expr = expr.to_sympy(lambda x: variable_generator(x, free), number_generator)
+            except ProtocolError as e:
+                print('  _[]_')
+                print('W (")')
+                print('|-(:\')-<')
+                print('|(\'  )')
+                print('JON HOW DO I RAISE A WARNING? ' + e.short_message)
```

Comment:
pycml used to just print to stderr. Using the logging module would be a nicer approach!

Reply:
OK someone can do that as part of #12 then